go/vt/sqlparser/sql.y

```diff
@@ -4876,6 +5098,28 @@ alter_table_statement_part:
   {
     $$ = &DDL{Action: AlterStr}
   }
+| AUTO_INCREMENT equal_opt expression
+  {
+    $$ = &DDL{Action: AlterStr, AutoIncSpec: &AutoIncSpec{Value: $3}}
+  }
+| partition_operation
+  {
+    $$ = &DDL{Action: AlterStr, PartitionSpec: $1}
+  }
+
+with_or_without:
+  WITH
+  {
+    $$ = true
+  }
+| WITHOUT
+  {
+    $$ = false
+  }
+
+any_non_reserved:
```

Comment:
This is probably better as `id_or_non_reserved_keyword`

Reply:
Will rename